test/performance/odp_ipsecfwd.c

```diff
@@ -1023,18 +1353,31 @@ static void create_sa_entry(uint32_t dir, uint32_t spi, const char *src_ip_str,
 	++config->num_sas;
 }
 
-static void parse_sas(prog_config_t *config)
+static void parse_and_create_sa_entries(config_t *cfg, prog_config_t *config, uint32_t max_num_sa)
+{
+	parse_sa_entry(cfg, "default.", &config->ovr_cfg);
+
+	for (uint32_t i = 0U; i < MAX_SAS; i++) {
+		char path[strlen("saXXXX.") + 1U];
+		sa_config_t sa_cfg;
+		int val;
+
+		odp_ipsec_sa_param_init(&sa_cfg.sa_param);
+		memset(&sa_cfg.sa_param.crypto, 0, sizeof(sa_cfg.sa_param.crypto));
+		snprintf(path, sizeof(path), "sa%u.", i);
```

Comment:
Perhaps it would be be better to have SAs inside a list in the config file so that no special naming would be needed and this code could just iterate over the list (using config_setting_get_elem()) instead of trying to read every possible sa config group? Similarly, fwd entries could be in a list.

Reply:
Swapped to list(s) in v3.